plugins/Live/javascripts/visitorProfile.js

```diff
@@ -155,8 +155,9 @@
             // append token_auth dynamically to export link
             $element.on('mousedown', '.visitor-profile-export', function (e) {
                 var url = $(this).attr('href');
-                if (url.indexOf('&token_auth=') == -1) {
-                    $(this).attr('href', url + '&force_api_session=1&token_auth=' + piwik.token_auth);
+                var augmentedUrl = piwik.broadcast.addTokenOrForceApiTo(url);
```

Comment:
here it would now be adding a token from the URL when it should be using `piwik.token_auth`. In some places further down (eg in  `plugins/Overlay/templates/index_noframe.twig`) it's no longer respecting `piwik.shouldPropagateTokenAuth`. This could cause various issues depending on configuration and usage etc (eg if someone was to POST the token_auth or if someone has `enable_framed_pages=1`). Currently, it breaks for example the visitor profile export when you go to "Visitor log -> Visitor profile -> Export". Ideally we wouldn't change any logic or behaviour to not break something and only refactor the code to remove some duplicate code.

As the token_auth may come from different places it might be good to make the method like

```js
broadcast.addTokenAuthToUrl = function(url, token_auth) {
    url += '&token_auth='  + encodeURIComponent(token_auth);
    if (!piwik.broadcast.isWidgetizeRequestWithoutSession()) {
        url += '&force_api_session=1';
    }
   return url;
}
```

 then it can be used from different sources like below and keep existing logic.

```js
var token_auth = piwik.broadcast.getValueFromUrl("token_auth");
if (token_auth.length && piwik.shouldPropagateTokenAuth) {
    url = piwik.broadcast.broadcast.addTokenAuthToUrl(url, token_auth);
}
```

If we're wanting to have the other code reusable
```js
        var token_auth = piwik.broadcast.getValueFromUrl("token_auth");
        if (token_auth.length && piwik.shouldPropagateTokenAuth) {
        return token_auth;
      }
```

could put this into a separate method maybe?

Reply:
Is this resolved, @tsteur ? If so can you resolve this conversation or let me know to do it?